### PR TITLE
UI cleanup and homeOwners validation edge case

### PR DESCRIPTION
**Pull request by owillborn**
*Issue #:* /bcgov/entity#13727

*Description of changes:*
-UI fixes (see original ticket for specifics)
-homeOwners section now validates when review and confirm screen is reached. Will display error bar on table if no owners added and review screen was reached.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <v-card flat class="py-6 px-8 rounded">
+  <v-card flat class="py-6 px-8 mb-20 rounded">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue]: `mb-20` is out of range of the helper classes. I think the max helper class is 10.

Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue]: To add, each increment here is worth about 4px by Vuetify definition. 
ie `mb-1` === 4px margin bottom and so on

Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/AddEditHomeOwner.vue]: Thanks for letting me know, was unaware!

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue [hunk 2]
```diff
@@ -343,6 +345,10 @@ export default defineComponent({
       }
     )
 
+    watch(() => context.root.$route.name, (route: string) => {
+      if (route === RouteNames.MHR_REVIEW_CONFIRM) localState.reviewed = true
+    })
+
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue]: I believe we already have a validation flag that is set when a user navigates to MHR_REVIEW_CONFIRM.

This line in the Review Confirm View, 
`setValidation(MhrSectVal.REVIEW_CONFIRM_VALID, MhrCompVal.VALIDATE_STEPS, true)` is already setting this global flag, so i don't believe you need localState `reviewed` here.

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue [hunk 3]
```diff
@@ -6,22 +6,25 @@
...
-      <section class="px-6 my-2" v-if="hasHomeOwners || (hasGroups && showGroups)">
+      <section v-show="!getStepValidation(MhrSectVal.HOME_OWNERS_VALID)"
+      :class="hasHomeOwners ? 'pt-30px px-6' : 'px-6 py-8'">
+        <span>
+          <v-icon color="error">mdi-information-outline</v-icon>
+          <span class="error-text mx-1">This step is unfinished.</span>
+          <router-link :to="{ path: `/${RouteNames.MHR_REGISTRATION}/${RouteNames.HOME_OWNERS}` }"
+          >Return to this step to complete it.</router-link>
+        </span>
+      </section>
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue]: Could you explain the change to `section` tag vs div? Is it based on validation scroll to effect?

Also refer to this for helper classes: https://vuetifyjs.com/en/styles/spacing/#how-it-works
Is `pt-30px` from vuetify or custom? 

Also, need an indent on the class binding `:class`

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue [hunk 4]
```diff
@@ -6,22 +6,25 @@
...
+
+      <section class="px-6 my-2" v-if="hasHomeOwners">
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: What displays here if this is `false` 
I guess previously there was fallback values below for missing information.

Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: If this is false only the error message will display, as per requirements in the ticket.
image

Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: ok fair enough, requirements maybe updated?

That being said, are the fallback values still utilized/required below?

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue (hunk 3 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue]: Switched to section simply for consistency with the other section tags within the div. Didn't see any visual differences, is one preferred over the other in this case?

Thanks for the helper class link, pt-30px is a custom class in base.scss

Good catch

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue (hunk 4 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: Not sure on what you mean by fallback values...

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue (hunk 2 above)
Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue]: You're right, I looked for a flag for a while but couldn't find it, guess its this one haha!

File: ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue (hunk 4 above)
Comment (cameron-eyds) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: Basically a fallback value is an `||` scenario. Ie render `someProperty` || 'N/A'

Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: If you mean the template within this v-if, then yes it is used to display the table on R/C when owners are present

Comment (owillborn) [reply ppr-ui/src/components/mhrRegistration/ReviewConfirm/HomeOwnersReview.vue:19]: ohhhh gotcha, then I guess there's no case in which tenancy type can be N/A but still have an owner present. I'll remove the fallbacks

**Review by dimak1 — APPROVED** [review]
Other than Cameron's comments, looks good to me.

**Review by cameron-eyds — APPROVED** [review]
Thanks for the updates!

**Pull request merged**